### PR TITLE
Fix link format for changes in PR 4940 (std.numeric.gcd)

### PR DESCRIPTION
**Pull request by wilzbach**
Fixup for #4940, which was merged prematurely :/

### CODE REVIEW

File: std/numeric.d [hunk 1]
```diff
@@ -2572,8 +2572,8 @@ GapWeightedSimilarityIncremental!(R, F) gapWeightedSimilarityIncremental(R, F)
 
 /**
 Computes the greatest common divisor of $(D a) and $(D b) by using
-an efficient algorithm such as $(HTTPS en.wikipedia.org/wiki/Euclidean_algorithm, Euclid's)
-or $(HTTPS en.wikipedia.org/wiki/Binary_GCD_algorithm, Stein's) algorithm.
+an efficient algorithm such as $(LINK2 en.wikipedia.org/wiki/Euclidean_algorithm, Euclid's)
+or $(LINK2 en.wikipedia.org/wiki/Binary_GCD_algorithm, Stein's) algorithm.
  */
```
Comment (andralex) [inline std/numeric.d]: I don't see this necessary.

**Comment by andralex** [comment]
I don't mean to antagonize but please let's let this go. Thanks.

**Comment by andralex** [comment]
Actually I might be missing something - was there a problem with the way links were previously generated?

File: std/numeric.d (hunk 1 above)
Comment (CyberShadow) [inline std/numeric.d]: This patch is wrong, you need to add the protocol if you change the macro away from `HTTP`/`HTTPS`.

**Comment by wilzbach** [comment]
Ah damn - I just read the PR mail on my phone and didn't realize that the change was already part of the PR.
-> sorry for the noise ;/

(there's the bigger discussion of deprecating HTTP/HTTPS in favor of LINK2, but that's another day)

**Comment by andralex** [comment]
@wilzbach an epic discussion is occurring right now on irc

**Comment by CyberShadow** [comment]
@wilzbach @aG0aep6G Andrei and I had a quick debate on IRC about HTTP/HTTPS vs. LINK2 & co. and one strong argument in favor of HTTP/HTTPS is DDoc's special handling handling of : (i.e. when the URL starts at the beginning of the line). When text is rewrapped in the editor this can put the URL at the beginning of the line, which spontaneously breaks links.

An idea we came to is to change DMD's DDoc parsing to treat `:` specially (as section header terminator) only when it's followed by whitespace (incl. a newline). Breakage should be minimal. With this change, moving away from HTTP/HTTPS macros would become more feasible. This can also make inline CSS less painful.

**Comment by MartinNowak** [comment]
The perils of having reinvented source code documentation format... ;)

**Comment by aG0aep6G** [comment]
> @wilzbach @aG0aep6G Andrei and I had a quick debate on IRC about HTTP/HTTPS vs. LINK2 & co. and one strong argument in favor of HTTP/HTTPS is DDoc's special handling handling of : (i.e. when the URL starts at the beginning of the line). When text is rewrapped in the editor this can put the URL at the beginning of the line, which spontaneously breaks links.

I'm aware of that argument. I think it's rather weak, because HTTP/HTTPS still fail when the comment is indented somehow, which is common.

> An idea we came to is to change DMD's DDoc parsing to treat : specially (as section header terminator) only when it's followed by whitespace (incl. a newline).

[Old news](https://github.com/dlang/phobos/pull/4762#r77437215) ;) Sounds good to me, obviously.

**Comment by andralex** [comment]
FWIW I don't indent documentation.

**Comment by andralex** [comment]
@MartinNowak which one are you referring to at the time ddoc was introduced?

**Comment by aG0aep6G** [comment]
> FWIW I don't indent documentation.

Yes, you do: https://github.com/dlang/phobos/blob/85fcd4adad8f3ce014456bbf948f3a70895cc132/std/experimental/checkedint.d#L195-L197

**Pull request closed without merging**